src/editor/tools/place_critter_tool.cc

```diff
@@ -97,6 +95,51 @@ int32_t EditorPlaceCritterTool::handle_undo_impl(
 	return 0;
 }
 
-EditorActionArgs EditorPlaceCritterTool::format_args_impl(EditorInteractive& parent) {
-	return EditorTool::format_args_impl(parent);
+EditorActionArgs EditorPlaceCritterTool::format_args_impl() {
+	return EditorTool::format_args_impl();
+}
+
+std::string EditorPlaceCritterTool::format_conf_description_impl(const ToolConf& conf) {
+	const Widelands::Descriptions& descriptions = parent_.egbase().descriptions();
+	const Widelands::DescriptionMaintainer<Widelands::CritterDescr>& critter_descriptions =
+	   descriptions.critters();
+
+	std::string buf;
+	constexpr int max_string_size = 100;
+	auto first = conf.map_obj_types.begin();
+
+	for (std::list<Widelands::DescriptionIndex>::const_iterator p = conf.map_obj_types.begin();
+	     p != conf.map_obj_types.end() && buf.size() < max_string_size; p++) {
+		if (p != first) {
+			buf += " | ";
+		}
+		buf += critter_descriptions.get(*p).descname();
+	}
+
+	/** TRANSLATORS: An entry in the tool history list. */
+	return format(_("Critter: %1$s"), buf);
+}
+
+bool EditorPlaceCritterTool::save_configuration_impl(ToolConf& conf) {
+	int j = get_nr_enabled();
```

Comment:
IMHO it would make sense to now rewrite MultiSelect to use an `std::set` instead of a `vector`. Code blocks like this here look awkward and could be written so much more elegantly with a better interface. This PR would benefit from incorporating this refactoring.

Relatedly, I wonder if some of those `std::list`s could also be changed to `std::set` for nicer code. It would all fit well into this PR but is not a must-have…

Reply:
I changed MultiSelect to use a set, and it did simplify the code a lot. Also changed the tool_conf list to set. I think the one in toolhistory_tool must be a list, so they are listed from oldest to newest and the oldest get removed first. It looks like the ones in action_args must also be lists, because the items represent coordinates where the mapobjects or terrains should be restored on undo.